discord_bot/discord_bot.py

```diff
@@ -34,53 +36,94 @@ def load_config():
         with open(CONFIG_FILE_PATH) as f:
             return json.load(f)
 
-    def get_gateway(self):
+    def get_gateway(self, **kwargs):
         """
-        Caches a gateway value and retrieves a new URL
+        Caches a gateway value, authenticates, and retrieves a new URL
         :return: gateway URL
         """
-        r = requests.get(self.config["discord_gateway_endpoint"])
+        header = {
+            "headers": {
+                "Authorization": "Bot {}".format(self.config["handshake_identity"]["token"]),
+                "User-Agent": "DiscordBot (https://github.com/ShifuYee/Discord-Bot, 0.0.1)"
```

Comment:
Shouldn't hard-code the version, there's probably a way to pull the version number from `setup.py` using stuff from `setuptools` documentation.